// @mendix/extensions-api/dist/index.js
import { getComponentFramework as n } from "@mendix/component-framework";
import { getComponentFramework as d } from "@mendix/component-framework";
import { getModelAccessWithComponentProxy as o } from "@mendix/model-access-sdk";
function t(i) {
  const e = n(i);
  return {
    ui: {
      messageBoxes: e.getApi("mendix.MessageBoxApi"),
      tabs: e.getApi("mendix.TabApi"),
      panes: e.getApi("mendix.DockablePaneApi"),
      extensionsMenu: e.getApi("mendix.ExtensionsMenuApi"),
      preferences: e.getApi("mendix.PreferencesApi"),
      dialogs: e.getApi("mendix.DialogApi"),
      notifications: e.getApi("mendix.NotificationApi"),
      editors: e.getApi("mendix.EditorApi"),
      elementSelectors: e.getApi("mendix.ElementSelectorApi"),
      versionControl: e.getApi("mendix.VersionControlApi"),
      appExplorer: e.getApi("mendix.AppExplorerApi"),
      documents: e.getApi("mendix.DocumentEditorApi"),
      messagePassing: e.getApi("mendix.MessagePassingApi")
    },
    app: {
      files: e.getApi("mendix.AppFilesApi"),
      runtime: e.getApi("mendix.RuntimeControllerApi"),
      model: {
        domainModels: o(
          "mendix.DomainModelApi",
          "DomainModels$DomainModel",
          i
        ),
        pages: o(
          "mendix.PageApi",
          "Pages$Page",
          i
        ),
        constants: o(
          "mendix.ConstantApi",
          "Constants$Constant",
          i
        ),
        enumerations: o(
          "mendix.EnumerationApi",
          "Enumerations$Enumeration",
          i
        ),
        snippets: o(
          "mendix.SnippetApi",
          "Pages$Snippet",
          i
        ),
        buildingBlocks: o(
          "mendix.BuildingBlockApi",
          "Pages$BuildingBlock",
          i
        ),
        projects: e.getApi("mendix.ProjectApi"),
        moduleSettings: o(
          "mendix.ModuleSettingsApi",
          "Projects$ModuleSettings",
          i
        ),
        microflows: o(
          "mendix.MicroflowModelApi",
          "Microflows$Microflow",
          i
        ),
        customBlobDocuments: e.getApi("mendix.CustomBlobDocumentApi"),
        importMappings: o(
          "mendix.ImportMappingModelApi",
          "ImportMappings$ImportMapping",
          i
        ),
        exportMappings: o(
          "mendix.ExportMappingModelApi",
          "ExportMappings$ExportMapping",
          i
        ),
        jsonStructures: o(
          "mendix.JsonStructureModelApi",
          "JsonStructures$JsonStructure",
          i
        ),
        messageDefinitions: o(
          "mendix.MessageDefinitionModelApi",
          "MessageDefinitions$MessageDefinitionCollection",
          i
        ),
        xmlSchemas: o(
          "mendix.XmlSchemaModelApi",
          "XmlSchemas$XmlSchema",
          i
        ),
        workflows: o(
          "mendix.WorkflowApi",
          "Workflows$Workflow",
          i
        )
      }
    },
    network: {
      httpProxy: e.getApi("mendix.HttpProxyApi")
    }
  };
}
export {
  d as getComponentFramework,
  t as getStudioProApi
};
